src/js/pages/jobs/JobDetailPageContainer.js

```diff
@@ -18,170 +14,112 @@ export const DIALOGS = {
-      errorMsg
-    });
-  }
-
-  onMetronomeStoreJobDeleteSuccess() {
-    // TODO: remove the closeDialog here.
-    this.closeDialog();
-    this.context.router.push("/jobs");
-  }
-
-  onMetronomeStoreJobDetailError() {
-    this.setState({ errorCount: this.state.errorCount + 1 });
-  }
-
-  onMetronomeStoreJobDetailChange() {
-    this.setState({ errorCount: 0, isLoading: false });
-  }
+const disabledDialog$ = new BehaviorSubject(null);
+const disableDialog = () => {
+  disabledDialog$.next(DIALOGS.DESTROY);
+};
 
-  render() {
-    const jobTree = MetronomeStore.jobTree;
-    if (this.state.errorCount > 3) {
-      return <ErrorScreen jobTree={jobTree} />;
-    }
+const jobActionDialog$ = new BehaviorSubject(null);
 
-    if (this.state.isLoading) {
-      return <LoadingScreen jobTree={jobTree} />;
-    }
+const handleDestroyButtonClick = () => {
+  jobActionDialog$.next(DIALOGS.DESTROY);
+};
 
-    const job = MetronomeStore.getJob(this.props.params.id);
-    const props = {
-      handleEditButtonClick: this.handleEditButtonClick,
-      handleRunNowButtonClick: this.handleRunNowButtonClick,
-      handleDisableScheduleButtonClick: this.handleDisableScheduleButtonClick,
-      handleEnableScheduleButtonClick: this.handleEnableScheduleButtonClick,
-      handleDestroyButtonClick: this.handleDestroyButtonClick,
-      handleAcceptDestroyDialog: this.handleAcceptDestroyDialog,
-      closeDialog: this.closeDialog,
-      job,
-      jobTree,
-      errorMsg: this.state.errorMsg,
-      disabledDialog: this.state.disabledDialog,
-      jobActionDialog: this.state.jobActionDialog
-    };
+const handleEditButtonClick = () => {
+  jobActionDialog$.next(DIALOGS.EDIT);
+};
 
-    return <JobDetailPage {...this.props} {...props} />;
-  }
-}
-JobDetailPageContainer.contextTypes = {
-  router: routerShape
+const closeDialog = () => {
+  jobActionDialog$.next(null);
+  disabledDialog$.next(null);
 };
 
-export default JobDetailPageContainer;
+const getGraphQL = id =>
+  graphqlObservable(
+    gql`
+      query {
+        job(id: $id) {
+          id
+          name
+          path
+          command
+          schedules
+          cpus
+          description
+          mem
+          docker
+          disk
+          labels
+          jobRuns
+          json
+        }
+      }
+    `,
+    schema,
+    { id }
+  );
+
+export default componentFromStream(props$ => {
+  const id$ = props$.map(props => props.params.id).distinctUntilChanged();
+  const job$ = id$
+    .switchMap(id => getGraphQL(id))
+    .map(({ data: { job } }) => job);
+
+  return Observable.combineLatest([
+    job$,
+    props$,
+    jobActionDialog$,
+    disabledDialog$
+  ])
+    .map(([job, props, jobActionDialog, disabledDialog]) => {
+      const enhancedProps = {
+        ...props,
+        job,
+        jobActionDialog,
+        disabledDialog,
+        errorMsg: null,
+        handleRunNowButtonClick() {
+          MetronomeStore.runJob(job.id);
```

Comment:
This doesnt work :/ (nothing happens after I click on "run now")

Do we have a plan to proceed here? Not sure if this has to be fixed in this PR, but did you sync with @Fabs that his work is relatively easy to integrate here?

Reply:
Tried it and it works just fine on my machine

Reply:
It should work with the store, and I think we should ship it with the store, and not wait for the menu to be ready.